mypy/types.py

```diff
@@ -1162,23 +1162,45 @@ class TypeType(Type):
     # a generic class instance, a union, Any, a type variable...
     item = None  # type: Type
 
+    # this class should not be created directly
+    # use make method which may return either TypeType or UnionType
+    # this is to ensure Type[Union[A, B]] is always represented as Union[Type[A], Type[B]]
     def __init__(self, item: Type, *, line: int = -1, column: int = -1) -> None:
```

Comment:
Hmm this is pretty magical. My suggestion is to do something along these lines:

* Have a `__init__` as before, but have it only accept `Union[Any, Instance]` or similar in the signature. Thus it can't be called with a union argument. Not sure what the exact set of accepted types should be. Maybe the best thing would be to use `TypeInfo` instead of `Instance`, but that would likely be a larger change.
* Also provide a `make_normalized` static method that also accepts other kinds of types and call the normal constructor one or more times. It can also handle `CallableType`.